New feature to manage GRANT

ultraheroe opened the pull request:
### Description of the Change

➕ Added
- New feature to manage GRANT perms and privs
- Corresponding vars to use that feature
- Testing vars for this feature in molecule/group_vars
- Necesary templates

🛠️ Fixed
- Tasks with no_log set to False

🔁 Updated
- Readme
- Test requirements Ansible version to 5.2.0
- Task tags
- Clickhouse default version

### Benefits

Some fix, updates and a new feature to manage permissions and privileges

### Possible Drawbacks

There are some limitations / things to take care:
- When performing the GRANT actions to maintain the perms & privs clean a "general" REVOKE is performed before GRANTing
- There is an option to disable that before "GRANTS" clean up
- Statements Aliases are valid, but not handled at "ansible level" so this results in task making comparisions like `privileges: [DELETE]` vs `system.grants access_type = ALTER DELETE` (from ClickHouse), so we recommend set "un-aliased" perms and privs.
- When performing REVOKE or GRANT if a problem occurs may be result in unexpected / removed perms & privs in the ClickHouse DB ¡¡Be extra careful!!

### Applicable Issues
#26

ultraheroe committed:
Merge pull request #22 from idealista/develop

New release

santi-eidu committed:
Merge pull request #25 from idealista/develop

Fixed #23 avoid internal database being removed.

ultraheroe committed:
New feature to manage GRANT

➕ Added
- New feature to manage GRANT perms and privs
- Corresponding vars to use that feature
- Testing vars for this feature in molecule/group_vars
- Necesary templates

🛠️ Fixed
- Tasks with no_log set to False

🔁 Updated
- Readme
- Test requirements Ansible version to 5.2.0
- Task tags

ultraheroe committed:
Update Readme and fix tasks

🔁 Updated
- Readme whith some extra info about GRANTs

🛠️ Fixed
- Idempotence in command with SQL querys

ultraheroe committed:
Update tests

🔁 Updated
- Tests and add one more to addecuate to new molecule group_vars

frantsao commented on tasks/config/databases.yml:
```diff
@@ -4,38 +4,42 @@
   changed_when: False
   when: clickhouse_default_database is defined
-  no_log: False
+  no_log: True
```
As it can log sensible information (user/password) no_log must be maintained, I think. Same for rest of similar changes.

ultraheroe committed:
Add missing when for grants perms & privs

ablancolopez commented on templates/DDL/REVOKE.j2:
```diff
@@ -0,0 +1,23 @@
+{% for grant in actual_privs %}
+{% for priv in grant.access_type %}
+{% if (grant.user_name in item.to) or (grant.role_name in item.to)
```
As this if has a lot of conditions, I would rather use `{% set %}` to give it a proper meaning

ultraheroe replied:
Template updated!

ultraheroe committed:
Use jinja set in template and a fix

- Use of jinja set in REVOKE template
- Fix grant cleaning task if some vars aren't defined use default a value
-  Updated Clickhouse default version

ultraheroe committed:
Suitable tests to new default version
